dbDeleteRowByKey will safly exist if given empty dt

Tim113 opened the pull request:
Before if dbDeleateRowByKey was given an empty data table it would
still try and run, this would then give an MySQL error.  The function
will now test the dimentions of dt and safly return if there are no
rows to deleat.

Commit:
dbDeleteRowByKey will safly exist if given empty dt

Before if dbDeleateRowByKey was given an empty data table it would
still try and run, this would then give an MySQL error.  The function
will now test the dimentions of dt and safly return if there are no
rows to deleat.

AkhilGNair commented on tests/testthat/test.dbDeleteRowByKey.R:
```diff
@@ -30,5 +30,12 @@ test_that("Exactly 50000 rows are deleted", {
+test_that("The function will safly exit if given empty dt", {
+
+  # Expect there is no error
+  expect_error(dbDeleteRowByKey(db, "Letters", dt_letters[1 == 0]), NA)
```
`# Expect there is no error`
This expects an error?

Tim113 replied:
Yes, really stupid but the NA at the end of the line means that it is expecting the error that is NA, which is no error.  So it is not expecting an error, even though it says expect_error.

Tim113 replied:
Got it from http://stackoverflow.com/questions/10826365/how-to-test-that-an-error-does-not-occur

AkhilGNair commented on R/dbDeleteRowByKey.R:
```diff
@@ -1,5 +1,5 @@
 # Global variables for data.table column names
-if(getRversion() >= "2.15.1")  utils::globalVariables(c(".id", "tup"))
+if (getRversion() >= "2.15.1")  utils::globalVariables(c(".id", "tup"))
```
Hadley's style guide says no space after an `if`

Tim113 replied:
Rstudio and I disagree, but ok.

AkhilGNair replied:
Yeah my bad. Thanks

AkhilGNair commented on R/dbDeleteRowByKey.R:
```diff
@@ -12,6 +12,10 @@ if(getRversion() >= "2.15.1")  utils::globalVariables(c(".id", "tup"))
 dbDeleteRowByKey = function(con, name, dt) {
+
+  # If there are no rows in dt then return, there is nothing to delete.
+  if (dt[, .N == 0 ]) {return()}
```
`if(dt[, .N] == 0) { return() }`

AkhilGNair reviewed (changes requested):
Little things mostly. Not sure about that test though?

AkhilNairAmey commented on R/dbDeleteRowByKey.R:
```diff
@@ -1,5 +1,5 @@
 # Global variables for data.table column names
-if(getRversion() >= "2.15.1")  utils::globalVariables(c(".id", "tup"))
+if (getRversion() >= "2.15.1")  utils::globalVariables(c(".id", "tup"))
```
Rstudio disagrees? orly? where

Commit:
Styling changes to the code

AkhilNairAmey commented:
Build failed :( I guess that means `expect_equals` didnt work?

Commit:
Gave dbDeleateRowByKey a return when no rows are removed

Now when dbDeleateRowByKey is passed an empty dt it will return TRUE.
The tests have been updated to reflect this.
